### DISCUSSION TITLE
What interfaces do already exist?

### DISCUSSION THREAD

**Discussion by FlorianHeigl** [Seagate/cortx#369] [General] [answered]
So I read the Readme, FAQ and a bit more.
I understood cortx is fresh, and interesting.
I learned it'll excel in scalability and be a bit smarter-than average in dealing with various faults that others in the past might simply have labeled as "hard problems" and skipped the effort. Sounds great.

My first thought was "dear god, maybe it'll be a fast S3 replacement for AWS some day".
And from there I finally arrived at the question - ok, actually, so it's an object store but why is there no word further about access?

So, something for the FAQ is: 

* what is currently supported for access? S3? Is there a fake Fuse FS layer? HTTP? just a library?
* what is currently supported for management? how do you see failures? Usage? Do tools already exist?

That should be answerable by reading 60 seconds (no context switch) instead of fetching the VM image and testing everything. One can still point at the VM image in all examples so it stays "motivational" for the download :-)

Also, I found it sad to read that this comes from an EU funded project but no mention of that project. It's nice to see something practical come out of i.e. H2020, but one would like to know more.

**Comment by johnbent** [comment, marked as answer]
Great suggestion for the FAQ!  Currently we support S3 as well as direct access via the libmotr API.  We also are working on a file interface that does not use FUSE.
Motr repo: https://github.com/Seagate/cortx-motr
POSIX repo: https://github.com/Seagate/cortx-posix

In terms of the EU funded project, that's another great suggestion for the FAQ.  Here is a link to some early blog posts about the project: https://www.hpcwire.com/2016/11/29/seagate-led-sage-project-delivers-update-exascale/
https://www.seagate.com/news/news-archive/seagates-joins-historic-european-commission-master-pr/

**Reply by FlorianHeigl** [reply]
Hi,

You had the same link for motr and POSIX work. Also, it was a very helpful answer!

**Comment by huanghua78** [comment]
Please refer to the following doc:
/doc/be/CORTX_TUTORIAL.rst
